src/commands/register.ts

```diff
@@ -360,4 +364,37 @@ export default class ComponentRegister extends BaseCommand {
   public static getTagFromFlags(flags: any): string {
     return flags.environment ? `${ENV_TAG_PREFIX}${flags.environment}` : flags.tag;
   }
+
+  private async generateDependenciesWarnings(component_dependencies: Dictionary<string>, account_id: string) {
+    const existing_account_components: Dictionary<string> = {};
+    const { rows, _ } = (await this.app.api.get(`/accounts/${account_id}/components`)).data;
+    for (const row of rows) {
+      existing_account_components[row.component.name] = row.tag;
+    }
+
+    const register_warnings: string[] = [];
+    const latest_tag_warnings: string[] = [];
+    for (const key of Object.keys(component_dependencies)) {
+      if (!existing_account_components[key]) {
+        register_warnings.push(key); 
+      } 
+      if (component_dependencies[key] !== 'latest') {
+        latest_tag_warnings.push(key);
+      }
+    }
+
+    const warning = chalk.hex('#FFA500');
+    if (register_warnings.length > 0) {
+      this.log(warning(`\nMissing dependencies that need to be registered:`));
+      for (const component_name of register_warnings) {
+        this.log(warning(`  - ${component_name}`));
+      }
+    }
+    if (latest_tag_warnings.length > 0) {
+      this.log(warning(`\nThe following dependencies should have the 'latest' tag:`));
```

Comment:
If I register a component with the dependencies block

```yml
dependencies:
  architect-ryan/nonsense: test
  architect-ryan/some: 0.0.0
  architect-ryan/thing: latest
```

I get the warning

```
Missing dependencies that need to be registered:
  - architect-ryan/nonsense
  - architect-ryan/some
  - architect-ryan/thing

The following dependencies should have the 'latest' tag:
  - architect-ryan/nonsense
  - architect-ryan/some
```

but I specifically wanted to use the component versions `architect-ryan/nonsense:test` and `architect-ryan/some:0.0.0`, so I wouldn't expect an error here. Why does the `latest` tag need to be registered?

Reply:
Ok. I see the issue now. I'll make the changes.